kediatr-core/src/main/kotlin/com/trendyol/kediatr/CommandBus.kt

```diff
@@ -24,5 +24,19 @@ interface CommandBus {
      * @param T  any [Notification] subclass to publish
      */
     suspend fun <T : Notification> publishNotificationAsync(notification: T)
+
+    fun <TRequest, TResponse> processPipeline(pipelineBehaviors: Collection<PipelineBehavior>, request: TRequest, act: () -> TResponse): TResponse {
+        pipelineBehaviors.forEach { it.preProcess(request) }
+        val result = act()
+        pipelineBehaviors.forEach { it.postProcess(request) }
+        return result
+    }
+
+    suspend fun <TRequest, TResponse> processAsyncPipeline(pipelineBehaviors: Collection<AsyncPipelineBehavior>, request: TRequest, act: suspend () -> TResponse): TResponse {
```

Comment:
I do not have prior knowledge about java interoperability of the project. If we have desire for java interoperability, I guess we need to add consider https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-default/ annotation to generate a JVM default method for non-abstract Kotlin interface member.